opencog/reasoning/RuleEngine/rules/relex-to-logic/TOBE.scm

```diff
@@ -0,0 +1,73 @@
+(define TOBE
+	(BindLink
+		(ListLink
+			(TypedVariableLink
+				(VariableNode "$a-parse")
+				(TypeNode "ParseNode")
+			)
```

Comment:
All of these patterns are very similar, and are very verbose. They might be easier to read if compactified a bit, so e.g. try this:
```
(define (vardecl a b) (TypedVariableLink (VariableNode a)((TypeNode b)))
```
Then you could just say `(vardecl "$a-parse" "ParseNode")` in just one line.  Or for even more traditional computerese-style, swap a and b...